contracts/strategies/BaseStrategy/BaseStrategy.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+/**
+ * @notice Defines the abstract contract for voting algorithms on grants
+ * within a round. Any new voting algorithm would be expected to
+ * extend this abstract contract.
+ * Every BaseStrategy contract would be unique to RoundImplementation
+ * and would be deployed before creating a round
+ */
+abstract contract BaseStrategy {
+
+   // --- Data ---
+
+  /// @notice Round address
+  address public roundAddress;
+
+
+  // --- Modifier ---
+
+  /// @notice modifier to check if sender is round contract.
+  modifier isRoundContract() {
+    require(roundAddress != address(0), "error: voting contract not linked to a round");
+    require(msg.sender == roundAddress, "error: can be invoked only by round contract");
+    _;
+  }
+
+
+  // --- Core methods ---
+
+  /**
+   * @notice Invoked by RoundImplementation on creation to
+   * set the round for which the voting contracts is to be used
+   *
+   */
+  function init() external virtual {
```

Comment:
- A strategy may need to be initialized with some data so it may be good idea to add a `bytes calldata encodedParams` parameter. 
- Why not to use `initialize() external initializer`
- What about using OZ `Initializable._disableInitializers` on the `contructor` to prevent the implementations to be initialized

Reply:
Having an initializer with params for the strategy makes sense and that's how i updated the BaseStrategy.
![image](https://github.com/allo-protocol/allo-contracts/assets/22886639/ea04e5e6-aef0-44bf-af9f-3d9eeec611e2)

you will now have to pass the Strategy params on round creation in the registry and it can be used by overwriting the `_initialize(...)` function.

Any thoughts?

Reply:
makes sense :D